### ISSUE TITLE
Observatory: EarthLocation with more site-related information

### ISSUE THREAD

**Issue by ceyzeriat** [astropy/astropy#5256]
Hi there,

If one wants to access the sites database in astropy (which I understand are observatories), one uses a class called `SiteRegistry` (`coordinates/sites.py`), which returns 1) a list of names and 2) a list of objects `EarthLocation` (`coordinates/earth.py`), through the use of a mapping dictionary `_lowercase_names_to_locations`.

Now that's not cool if the database of observatories gathers not only `latitude`, `longitude` and `elevation`, but also the timezone, the average pressure, the average temperature, and any other relevant information (source, aliases, website, humidity, wind, etc).

I dirty fix is to access `EarthLocation._get_site_registry()._loaded_jsondb` which stores the raw content of the database, however that's not really satisfactory for many reasons.

The idea would be to implement a new class "Observatory" (or whatever better name), which gathers all observatory information. It would most probably inherit from `EarthLocation`.
There could be another new class "ObservatoryRegistry", that mimics `SiteRegistry` unless it is replaced to output a list of "Observatory" instead of a list of `EarthLocation`.

What do you think?

CC astroplan @cdeil, @bmorris3, @bsipocz

Cheers

**Comment by bmorris3** [comment]
Maybe we should start by taking a step back – what would these additional quantities enable you to do that you can't currently do with `EarthLocation` objects? 

I think the timezone is the additional quantity that is most useful of the ones that you listed, and also the least likely to change as a function of time. A good case could be made that we could use a _function_ that returns the timezone for a given `EarthLocation`, which would prevent the need for making a more complicated database of observatories. Though that is admittedly more difficult than just putting in the timezones by hand for each observatory in the database.

**Comment by ceyzeriat** [comment]
I really don't understand you guys.
1) the timezone information has already been added by myself for each of the observatories, see https://github.com/astropy/astropy-data/pull/16. Depending on the new
2) an observatory is quite unlikely to relocate and change timezone, so I don't see the point of having such a _Coords-to-Timezone_ functionnality for database obs. Such fct will need to query online services for geo-location hence take unnecessary process/ping time and require the user to have an active internet connection for an information that can be safely qualified of _permanent_ and saved within the database obs.
3) astronomical observation planning at observatories in the database represent probably 99% of the use-cases, for which the timezone can be frozen. For the remaining 1%, the user can either input manually the timezone, or save locally his personal observatory with the right timezone so that he doesn't have to input it every time.
The _Coords-to-Timezone_ functionality you want to achieve applies to the 1% of use-cases, where the user didn't bother to save his obs data locally.
Famous quote that sums up what I think you want to achieve: "If you wish to make apple pie from scratch, you must first create the universe". (C. Sagan)
In short, that's over-kill, and I'm not doing it.

Re other information to be included in the Observatory structure.
Maybe the pressure and temperature is not extremely important, although it plays a significant role for calculating the target elevation when one wants to include diffraction/refraction effects (e.g. near horizon).
The sun/moon rejection radius do play a role when selecting targets for bright nights or radio-astronomers; either here or somewhere else, that's something the observer must be aware of.
Over and out.
